src/Grid/LocalDataSource.ts

```diff
@@ -0,0 +1,332 @@
+          if (filterableColumn.DataType === 'datetime' ||
+              filterableColumn.DataType === 'date' ||
+              filterableColumn.DataType === 'datetimeutc') {
+                subset = subset.filter((row) =>
+                  moment(row[filterableColumn.Name]).isSameOrAfter(moment(filterableColumn.Filter.Text)) &&
+                  moment(row[filterableColumn.Name]).isSameOrBefore(moment(filterableColumn.Filter.Argument[0])));
+            } else {
+              subset = subset.filter((row) => row[filterableColumn.Name] >= filterableColumn.Filter.Text &&
+                  row[filterableColumn.Name] <= filterableColumn.Filter.Argument[0]);
+            }
+          break;
+        default:
+          throw new Error('Unsupported Compare Operator');
+      }
+    });
+
+    return subset;
+  }
+
+  public applySorting(request: any, subset: any[]) {
+    let sortedColumns = _.filter(request.Columns, (column) =>
+      column.SortOrder > 0);
+
+    if (sortedColumns.length > 0) {
+      sortedColumns = _.sortBy(sortedColumns, ['SortOrder']);
+
+      const columns: any[] = [];
+      const orders: any[] = [];
+
+      _.forEachRight(sortedColumns, (column) => {
+        columns.push(column.Name);
+
+        orders.push((column.SortDirection === ColumnSortDirection.ASCENDING ? 'asc' : 'desc'));
+      });
+
+      subset = _.orderBy(subset, columns, orders);
+    } else {
+      subset = _.orderBy(subset, request.Columns[0].Name, 'asc');
+    }
+
+    return subset;
+  }
+
+  public getAggregatePayload(request: any, subset: any[]) {
+    const aggregateColumns = _.filter(request.Columns, (column) =>
+      column.Aggregate && column.Aggregate.toLowerCase() !== AggregateFunctions.NONE.toLowerCase());
+
+    const results = _.map(aggregateColumns, (column) => {
+      let value;
+
+      switch (column.Aggregate.toLowerCase()) {
+        case AggregateFunctions.SUM.toLowerCase():
+          value = _.sumBy(subset, column.Name);
+          break;
+        case AggregateFunctions.AVERAGE.toLowerCase():
+          value = _.meanBy(subset, column.Name);
+          break;
+        case AggregateFunctions.MAX.toLowerCase():
+          value = _.maxBy(subset, column.Name)[column.Name];
+          break;
+        case AggregateFunctions.MIN.toLowerCase():
+          value = _.minBy(subset, column.Name)[column.Name];
+          break;
+        case AggregateFunctions.COUNT.toLowerCase():
+          value = subset.length;
+          break;
+        case AggregateFunctions.DISTINCT_COUNT.toLowerCase():
+          value = _.uniqWith(subset, (a, b) => {
+            return a[column.Name] === b[column.Name];
+          }).length;
+          break;
+        default:
+          throw new Error('Unsupported aggregate function');
+      }
+
+      return { [column.Name]: value };
+    });
+
+    return _.reduce(results, _.merge, {});
+  }
+
+  /** Private methods */
+
+  private normalizeColumns = (columns: ColumnModel[]) =>
```

Comment:
Duplicated method... there must be a way to avoid having this method on every Data Source